scipy/special/_logsumexp.py

```diff
@@ -104,19 +104,43 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
     elif not np.isfinite(a_max):
         a_max = 0
 
+    tmp0 = (a == a_max) # compute largest separately
+
+
     if b is not None:
         b = np.asarray(b)
-        tmp = b * np.exp(a - a_max)
+        tmp = b * np.exp(a - a_max)  * (~tmp0)
+        tmp0 = b*(tmp0 + .0)
```

Comment:
This is to convert a bool array to float? Prefer more explicit `tmp0 = (a == a_max).astype(float)` or some such. 

Also might want to make it a bit more explicit with boolean indexing, like `mask = a == a_max`, `a1 = a[~mask]` etc.